Source/Core/S2Cell.js

```diff
@@ -0,0 +1,589 @@
+  while (cellId !== BigInt(0)) {
+    // eslint-disable-next-line
+    if (cellId & BigInt(1)) {
+      break;
+    }
+    lsbPosition++;
+    cellId = cellId >> BigInt(1); // eslint-disable-line
+  }
+
+  return S2MaxLevel - (lsbPosition >> 1);
+};
+
+/**
+ * Gets the child cell of the cell at the given index.
+ *
+ * @param {Number} index An integer index of the child.
+ * @returns {S2Cell} The child of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getChild = function (index) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.number("index", index);
+  if (index < 0 || index > 3) {
+    throw new DeveloperError("child index must be in the range [0-3].");
+  }
+  if (this._level === 30) {
+    throw new DeveloperError("cannot get child of leaf cell.");
+  }
+  //>>includeEnd('debug');
+
+  var newLsb = lsb(this._cellId) >> BigInt(2); // eslint-disable-line
+  var childCellId = this._cellId + BigInt(2 * index + 1 - 4) * newLsb; // eslint-disable-line
+  return new S2Cell(childCellId);
+};
+
+/**
+ * Gets the parent cell of an S2Cell.
+ *
+ * @returns {S2Cell} Returns the parent of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getParent = function () {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._level === 0) {
+    throw new DeveloperError("cannot get parent of root cell.");
+  }
+  //>>includeEnd('debug');
+  var newLsb = lsb(this._cellId) << BigInt(2); // eslint-disable-line
+  return new S2Cell((this._cellId & (~newLsb + BigInt(1))) | newLsb); // eslint-disable-line
+};
+
+/**
+ * Gets the parent cell at the given level.
+ *
+ * @returns {S2Cell} Returns the parent of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getParentAtLevel = function (level) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._level === 0 || level < 0 || this._level < level) {
+    throw new DeveloperError("cannot get parent at invalid level.");
+  }
+  //>>includeEnd('debug');
+  var newLsb = lsbForLevel(level);
+  return new S2Cell((this._cellId & -newLsb) | newLsb);
+};
+
+/**
+ * @private
+ */
+S2Cell.fromFacePosLevel = function (face, pos, level) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.bigint("pos", pos);
+  if (face < 0 || face > 5) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+
+  if (level < 0 || level > S2MaxLevel) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+  if (pos < 0 || pos > Math.pow(4, level)) {
+    throw new DeveloperError("Invalid Hilbert position for level");
+  }
+  //>>includeEnd('debug');
+  var cell = new S2Cell((face << BigInt(S2PositionBits)) + (pos | BigInt(1))); // eslint-disable-line
+  return cell.getParentAtLevel(level);
+};
+
+/**
+ * Get center of the S2 cell.
+ *
+ * @returns {Cartesian} The position of center of the S2 cell.
+ * @private
+ */
+S2Cell.prototype.getCenter = function () {
+  var center = getS2Center(this._cellId);
+  // Normalize XYZ.
+  Cartesian3.normalize(center, center);
```

Comment:
Again, do the idiomatic thing, even if it feels a bit verbose:

```suggestion
  center = Cartesian3.normalize(center, center);
```